src/main/webapp/app/exercises/text/assess/text-assessment-dashboard/text-assessment-dashboard.component.html

```diff
@@ -116,6 +117,7 @@ <h4 jhiTranslate="artemisApp.textAssessment.dashboard.heading" [translateValues]
                                     submission.id,
                                     'assessment'
                                 ]"
+                                [queryParams]="{ 'correction-round': submission.results?.length - 1 }"
```

Comment:
`submission.results?.length - 1`: Won't this evaluate to `-1` if there is no results object set (or no result present)? If we are sure there is at least one result we should use `!` instead of `?`.

Reply:
yes we should indeed change it to a !. When this button is pressed one result will already exist.